src/c-writer.cc

```diff
@@ -3387,6 +3542,30 @@ Result CWriter::WriteModule(const Module& module) {
   return result_;
 }
 
+// static
+const char* CWriter::GetReferenceTypeName(const Type& type) {
+  switch (type) {
+    case Type::FuncRef:
+      return "funcref";
+    case Type::ExternRef:
+      return "externref";
+    default:
+      WABT_UNREACHABLE;
+  }
+}
+
+// static
+const char* CWriter::GetReferenceNullValue(const Type& type) {
+  switch (type) {
+    case Type::FuncRef:
+      return "(wasm_rt_funcref_t){0, NULL, 0}";
```

Comment:
Can we define `wasm_rt_funcref_null_value` too?   Also, can we avoid having `wasm_rt_externref_null_value` in the wasm-rt.h header file if its only  used here in the generated code?

Reply:
Sure, done (wasm_rt_funcref_null_value). We wanted wasm_rt_externref_null_value to be in the wasm-rt.h header so that if the embedder wants to choose a different type for wasm_rt_externref_t (something other than a void*), then the embedder can also specify the appropriate null value. (In Fix we're using externref = m256 instead of void*.)